supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -401,9 +431,41 @@ class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
 
   override def requestsAbortAndDiesImmediately: Boolean = false
 
+  override def onTaskComplete(runStatus: TesRunStatus, handle: StandardAsyncPendingExecutionHandle): Unit = {
+    val taskEndTime = getTaskEndTime(handle, getTaskLogs)
+    if (runStatus == Error() | runStatus == Failed()) {
+      val errors = getErrorSeq(runStatus, handle, getErrorLogs)
+      errors.onComplete {
+        case Success(r) => tellMetadata(Map(CallMetadataKeys.Failures -> r))
+        case Failure(e) => log.error(e.getMessage)
+      }
+    }
+
+    taskEndTime.onComplete {
+      case Success(result) => tellMetadata(Map(CallMetadataKeys.TaskEndTime -> result))
+      case Failure(e) => log.error(e.getMessage)
+    }
+  }
+
   override def pollStatusAsync(handle: StandardAsyncPendingExecutionHandle): Future[TesRunStatus] =
```

Comment:
Could you write a comment describing the desired polling behavior here? The logic is a little complicated (for good performance reasons!), and future me would appreciate a description of the intended behavior.